### PR TITLE
Converting set to sorted sequence

### PR DESCRIPTION
**Pull request by stefan-aws**
Adding support for set to ordered sequence conversion, as proposed by @cpitclaudel in https://github.com/dafny-lang/libraries/issues/59. I changed some of his initial notation.

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

**Comment by davidcok** [comment]
Suggestion: Mark this module as function-syntax 4 and use Dafny4 syntax, so the text does not need to be changed later.

**Review by atomb — DISMISSED** [review]
This looks nice. One optional change I might make, but that's totally a matter of taste, would be to rename `GhostSetToSeq` to `SetToSeqSpec`.

**Review by atomb — APPROVED** [review]

File: src/Collections/Sequences/Seq.dfy [hunk 1]
```diff
@@ -770,4 +774,66 @@ module {:options "-functionSyntax:4"} Seq {
...
+      invariant multiset(left) + multiset(xs) == multiset(s)
+    {
+      var x :| x in left;
+      left := left - {x};
+      xs := xs + [x];
+    }
+  }
+
+  /* Proves that any two sequences that are sorted by a total order and admit the
+     same multiset are equal. */
```
Comment (MikaelMayer) [inline src/Collections/Sequences/Seq.dfy:807]:
```suggestion
  /* Proves that any two sequences that are sorted by a total order and that have the same elements are equal. */
```

File: src/Collections/Sequences/Seq.dfy [hunk 2]
```diff
@@ -770,4 +774,66 @@ module {:options "-functionSyntax:4"} Seq {
...
+      assert xs == [xs[0]] + xs[1..];
+      assert ys == [ys[0]] + ys[1..];
+      assert multiset(xs[1..]) == multiset(xs) - multiset{xs[0]};
+      assert multiset(ys[1..]) == multiset(ys) - multiset{ys[0]};
+      assert multiset(xs[1..]) == multiset(ys[1..]);
+      SortedUnique(xs[1..], ys[1..], R);
+    }
+  }
+
+  /* Converts a set to sequence that is ordered w.r.t. a given total order. */
```
Comment (MikaelMayer) [inline src/Collections/Sequences/Seq.dfy:827]:
```suggestion
  /* Converts a set to a sequence that is ordered w.r.t. a given total order. */
```

**Review by MikaelMayer — CHANGES_REQUESTED** [review]
Great addition, two minor comments

**Review by MikaelMayer — APPROVED** [review]
Great job !

**Review by atomb — APPROVED** [review]

**Pull request merged**